client/core/core.go

```diff
@@ -1071,6 +1052,125 @@ func (c *Core) ConnectWallet(assetID uint32) error {
+	}
+	return dbWallet.Settings, nil
+}
+
+// ReconfigureWallet updates the wallet configuration settings.
+func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]string) error {
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return newError(authErr, "ReconfigureWallet password error: %v", err)
+	}
+	c.walletMtx.Lock()
+	defer c.walletMtx.Unlock()
+	oldWallet, found := c.wallets[assetID]
+	if !found {
+		return newError(missingWalletErr, "%d -> %s wallet not found", assetID, unbip(assetID))
+	}
+	dbWallet := &db.Wallet{
+		AssetID:     oldWallet.AssetID,
+		Settings:    cfg,
+		Balance:     oldWallet.balance,
+		EncryptedPW: oldWallet.encPW,
+		Address:     oldWallet.address,
+	}
+	// Reload the wallet with the new settings.
+	wallet, err := c.loadWallet(dbWallet)
+	if err != nil {
+		return newError(walletErr, "error loading wallet for %d -> %s: %v", assetID, unbip(assetID), err)
+	}
+	// Must connect to ensure settings are good.
+	err = wallet.Connect(c.ctx)
+	if err != nil {
+		return newError(connectErr, "error connecting wallet: %v", err)
+	}
+	if oldWallet.unlocked() {
+		err := unlockWallet(wallet, crypter)
+		if err != nil {
+			wallet.Disconnect()
+			return newError(walletAuthErr, "wallet successfully connected, but errored unlocking. reconfiguration not saved: %v", err)
+		}
+	}
+	err = c.db.UpdateWallet(dbWallet)
+	if err != nil {
+		wallet.Disconnect()
+		return newError(dbErr, "error saving wallet configuration: %v", err)
+	}
+
+	c.connMtx.RLock()
+	for _, dc := range c.conns {
+		dc.tradeMtx.RLock()
+		for _, tracker := range dc.trades {
+			tracker.mtx.Lock()
+			if tracker.wallets.fromWallet.AssetID == assetID {
+				tracker.wallets.fromWallet = wallet
+			} else if tracker.wallets.toWallet.AssetID == assetID {
+				tracker.wallets.toWallet = wallet
+			}
+			tracker.mtx.Unlock()
+		}
+		dc.tradeMtx.RUnlock()
+	}
+	c.connMtx.RUnlock()
+
+	if oldWallet.connected() {
+		oldWallet.Disconnect()
+	}
+	c.wallets[assetID] = wallet
+
+	details := fmt.Sprintf("Configuration for %s wallet has been updated.", unbip(assetID))
+	c.notify(newWalletConfigNote("Wallet Configuration Updated", details, db.Success, wallet.state()))
+
+	return nil
+}
+
+// SetWalletPassword updates the (encrypted) password for the wallet.
+func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) error {
+	// Check the app password and get the crypter.
+	crypter, err := c.encryptionKey(appPW)
+	if err != nil {
+		return newError(authErr, "SetWalletPassword password error: %v", err)
+	}
+
+	// Check that the specified wallet exists.
+	c.walletMtx.Lock()
+	defer c.walletMtx.Unlock()
+	wallet, found := c.wallets[assetID]
+	if !found {
+		return newError(missingWalletErr, "wallet for %s (%d) is not known", unbip(assetID), assetID)
+	}
+
+	// Encrypt the password.
+	encPW, err := crypter.Encrypt(newPW)
+	if err != nil {
+		return newError(encryptionErr, "encryption error: %v", err)
+	}
+
```

Comment:
I think there should be a check here to make sure the new password can unlock the wallet. Otherwise, the notice that their wallet cannot be unlocked may come much later.

Reply:
I considered it, but my reasoning was that a user should be able to change their configuration settings even if their wallet software is not running. Thoughts?

Reply:
You cannot currently create a new wallet if the password is wrong, or if the wallet isn't connected. You also cannot change the settings (like account name, rpc user/pw) after the changes in this pr if the wallet is unable to connect. Only changing the password would not have that connected/running requirement then. I think it would be ok to require that.